Documentation/app-operations.md

```diff
@@ -0,0 +1,225 @@
+
+The responsibility of this entrypoint is to receive an app inside a pod and to remove it from the pod.
+so that it is ready to be started by the stage1 supervisor (e.g. systemd for the default flavor).
+After this operation, starting the application again is not possible and requires injecting the same
+application using the `app/inject` entrypoint.
+
+### Stage1 Default Flavor Example Workflow
+
+1. Stop the application using the `app stop` entrypoint.
+2. Delete the application rootfs in `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
+3. Remove the application service files.
+4. Perform garbage collection during the execution of a pod.
+5. Reload the systemd daemon for changes to take place without disturbing the other tennant applications.
+
+## rkt app start
+
+`coreos.com/rkt/stage1/app/start`
+
+1. given a reference to an application that's in a startable state (see below), instruct the supervisor to start the application.
+
+The responsibility of this entrypoint is to start an application that is in a `Ready` or `Stopped` state, 
+which might be an app that was recently injected or an app that was stopped, instruct the supervisor to start the application.
+
+## rkt app stop
+
+`coreos.com/rkt/stage1/app/stop`
+
+1. given a reference to an application that's in a stoppable state (see below), instruct the supervisor to stop the application.
+
+The responsibility of this entrypoint is to stop an application that is `Running` by instructing the stage1 supervisor. 
+
+**Note:** there is a graceful shutdown which sends the termination signal to application and waits for a grace period
+for application to terminate and then the application is shutdown if it doesn't terminate by the end of the grace period.
+
+**Discussion Point:** Should one-app-pods not be allowed to terminate when the app inside is stopped?
+At the moment, once the app stops the reaper terminates and the pod shuts down.
+
+# App Container States
+
+Expected set of application container states are listed below:
+
+```go
+type AppState string
+
+const (
+	// Ready to be started
+	ReadyAppState   AppState = "Ready"
+	StoppedAppState AppState = "Stop"
+
+	// Started
+	RunningAppState AppState = "Running"
+
+	FinishedAppState AppState = "Finished"
+
+	// No further operations can be done
+	EjectedAppState AppState = "Ejected"
+)
+```
+
+# Use Cases
+
+## Low-level  Pods
+
+Grant granular access to pods for orchestration systems and allow orchestration systems to develop their
+own pod concept on top of the exposed app-level operations.
+
+### Workflow
+
+1. Create an empty pod (simulated via a pod that contains a dumb ACI sleeping forever).
+2. Inject applications into the pod.
+3. Orchestrate the workflow of applications (e.g. app1 has to terminate successfully before app2).
+
+## Pod Mutability
+
+Enable in-place updates of a pod without disrupting the operations of the pod.
+
+### Workflow
+
+1. Inject updated applications that are meant to replace the old images.
```

Comment:
The basic action "inject" is to add a new application in the pod, right? Not necessary with the goal of replacing another image?